.github/workflows/test.yml

```diff
@@ -47,25 +50,26 @@ jobs:
           poetry run python main.py
     - name: Run the rust test cases to read the python artifacts
       run: cargo run -p compare-ref-impl-tests
+
   lint:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
+    - uses: actions/checkout@v4
+    - uses: actions-rust-lang/setup-rust-toolchain@v1
       with:
-        toolchain: nightly
-        override: true
         components: rustfmt, clippy
     - name: Run lint check
       run: cargo clippy --all --all-targets -- -D warnings
+
   format:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
+    - uses: actions/checkout@v4
+    - uses: actions-rust-lang/setup-rust-toolchain@v1
       with:
+        # We need nightly for rustfmt's ignore option. Without this rustfmt will
+        # complain about our generated code.
         toolchain: nightly
-        override: true
```

Comment:
What exactly does `rustfmt` complain about? Perhaps it is something we can fix/whitelist?

Reply:
The generated ANTLR parser. We could format that and commit the formatted result if you prefer.

Reply:
I've submitted #86 to improve the rustfmt situation.